### ISSUE TITLE
Support server-side dependency build for Linux

### ISSUE THREAD

**Issue by asavaritayal** [Azure/Azure-Functions#1034]
**Python customer context:**
Currently, users have no way of publishing Python Functions to Azure that use native dependencies. This is especially important for Machine Learning scenarios that use native C extension packages that don't support manylinux wheels for Python 3.6 on PyPI. 
Supporting a server-side build will unblock this scenario and enable a faster publishing time / smaller deployment payload.

**Node customer context**
@ColbyTresness

**Comment by ColbyTresness** [comment]
I'm unsure how much of this can be solved by better integration with Azure DevOps, but this is also critical for Node customers (for both Windows and Linux.) Customers can already build and deploy locally, but that falls flat in any more comprehensive scenario with source control involved, since Kudu is insufficient for all but the most simple npm installs. So I'll just emphasize that this is important across the board, not just for Python and Linux.

**Comment by asavaritayal** [comment]
Agreed, I'm removing the label Python but leaving Linux since the Kudu capability simply does not exist in Linux consumption today, which is a functionality gap for our users.
